### DISCUSSION TITLE
System.IO.IOException : The process cannot access the file 'xxx.dll' because it is being used by another process.

### DISCUSSION THREAD

**Discussion by Quaverflow** [jbevain/cecil#838] [Q&A] [answered]
Hi Jb,

![cecilissue](https://user-images.githubusercontent.com/81313844/154927365-a3f3f86d-618e-41e8-b8b8-99cce7ec9b82.png)

I've been trying to use Cecil to hook methods on the executing
assembly (see attached picture), just to start getting used to the
framework. Unfortunately, no matter what I do, I can't get it to work.
Using the ReadWrite = true will throw an exception.
Not using ReadWrite will not insert the new method call before the old one.

Using XUnit and .net6

Could you please point me in the right direction?
Thank you in advance.

Cheers,
Mirko

**Comment by jbevain** [comment, marked as answer]
The problem here is that you're trying to modify the binary which is currently being executed. Unfortunately, this cannot work.

**Reply by Quaverflow** [reply]
Ah I see! I was trying to play around with some injection in the way that HarmonyLib and MonoMod do. I'll have to dig around some more. Thabk you so much!
